### PR TITLE
Server-side rendering, with two options

### PR DESCRIPTION
**Pull request by samliebow**
Several major changes here:
1. Node clustering to improve performance.
2. Addition of Artillery test file and New Relic monitoring.
3. Removal of React and ReactDOM as static assets; using CDN links instead.
4. Automatic fetching of bundles from service servers. (Currently only tested on my own component, hence the other paths being commented out in server/serviceServers.js.)
5. Server-side rendering! 
- One pair of files (server/serverDataServerside.js and templates/templateDataServerside.js) requests data from microservices on the server side before sending over the complete page, so that rehydration only adds interactivity (and components don't request data on mount). 
- The other pair (server/serverDataClientside.js and templates/templateDataClientside.js) sends over components rendered with empty placeholder data, which request data on rehydration. 

From now on I'm planning to work exclusively with client-side data requests, for proxy performance speed and ease of intercompatibility with others' services.

### CODE REVIEW

File: templates/templateDataServerside.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    <body>
+    ${body}
+      <script crossorigin src="https://unpkg.com/react@16/umd/react.development.js"></script>
+      <script crossorigin src="https://unpkg.com/react-dom@16/umd/react-dom.development.js"></script>
+      ${servicesData.map(([serviceName]) => `<script src="/services/${serviceName}.js"></script>`).join('\n')}
+      <script>
+        ${servicesData.map(([serviceName, data])=> `
+          const props = JSON.parse(\`${JSON.stringify(data)}\`);
+          console.log(props);
+          ReactDOM.hydrate(
```
Comment (2000prcs) [inline templates/templateDataServerside.js]: Remember to remove unnecessary console.log!

File: artillery.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    - duration: 60
+      arrivalRate: 20
+
+scenarios:
+  - name: "Look at a random project"
+    flow:
+      - loop:
+        - get:
+            url: "/?{{ $randomNumber(1,10000000) }}"
+        count: 250
```
Comment (2000prcs) [inline artillery.yml:13]: Why are you looping a lot here?

**Review by 2000prcs — COMMENTED** [review]
Overall the code looks good! Just left few comments.

File: templates/templateDataServerside.js (hunk 1 above)
Comment (samliebow) [reply templates/templateDataServerside.js]: Thanks, fixed.

File: artillery.yml (hunk 2 above)
Comment (samliebow) [reply artillery.yml:13]: Looping a ton allowed me to test the maximum RPS of my server (pre-SSR) without creating a bunch of computationally expensive virtual users.

**Pull request merged**